src/Features/CSharp/CodeRefactorings/InlineTemporary/InlineTemporaryCodeRefactoringProvider.cs

```diff
@@ -514,7 +514,9 @@ private static SyntaxNode GetTopMostParentingExpression(ExpressionSyntax express
                 var originalNode = originalNodesEnum.Current;
 
                 // expressionToInline is Parenthesized prior to replacement, so get the parenting parenthesized expression.
-                var inlinedNode = (ExpressionSyntax)inlinedNodesEnum.Current.AsNode().Parent;
+                var currentInlinedNode = inlinedNodesEnum.Current;
+
+                var inlinedNode = (ExpressionSyntax)(currentInlinedNode.IsNode ? currentInlinedNode.AsNode().Parent : currentInlinedNode.AsToken().Parent);
```

Comment:
currentInlinedNode is a SyntaxNodeOrToken and should then be named "currentInlinedNodeOrToken". SyntaxNodeOrToken has a Parent property, so you can just use that Parent property directly without the AsNode()/AsToken() conversions. It seems like you can keep the original code and just remove the ".AsNode()"